### PR TITLE
feat(breadcrumbs): Add drawer for breadcrumbs

### PR DESCRIPTION
**Pull request by leeandher**
This PR adds a drawer and new control interface for breadcrumbs when expanded. The new UI doesn't allow for scrolling in the breadcrumbs, and instead provides a few of them as a summary always sorted newest -> oldest (since this will show the event associated with the issue)

image

**todo**
- [x] Add tests

### CODE REVIEW

File: static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx [hunk 1]
```diff
@@ -1,83 +1,88 @@
...
   const breadcrumbs: RawCrumb[] = useMemo(
     () => event.entries?.[breadcrumbEntryIndex]?.data?.values ?? [],
     [event, breadcrumbEntryIndex]
   );
-  const allCrumbs = useMemo(() => [...breadcrumbs], [breadcrumbs]);
   // Mapping of breadcrumb index -> breadcrumb meta
   const meta: Record<number, any> =
     event._meta?.entries?.[breadcrumbEntryIndex]?.data?.values;
 
+  let allCrumbs = useMemo(() => [...breadcrumbs], [breadcrumbs]);
```
Comment (scttcper) [inline static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx:57]: i know you're just moving this but what is this memoing

Comment (scttcper) [inline static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx:57]: useMemo'd things should be constants

File: static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx [hunk 2]
```diff
@@ -146,35 +139,34 @@ export default function BreadcrumbsDataSection({event}: BreadcrumbsDataSectionPr
...
-              breadcrumbs={searchedCrumbs}
-              virtualCrumbIndex={virtualCrumbIndex}
-              meta={meta}
-              sort={sort}
-              timeDisplay={timeDisplay}
-            />
-          </ScrollBox>
+        {allCrumbs.length ? (
+          <BreadcrumbsTimeline
+            breadcrumbs={summaryCrumbs}
```
Comment (scttcper) [inline static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx:148]: if summaryCrumbs is the thing getting rendered, it should probably be what the useMemo is returning

File: static/app/components/events/breadcrumbs/breadcrumbsDrawerContent.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        size="xs"
+        trigger={(props, isOpen) => (
+          <BreadcrumbControl
+            size="xs"
+            icon={<IconClock />}
+            showChevron={false}
+            isOpen={isOpen}
+            {...props}
+          />
+        )}
```
Comment (scttcper) [inline static/app/components/events/breadcrumbs/breadcrumbsDrawerContent.tsx:153]: i think you might be able to do this with triggerProps `triggerProps={{icon:  <IconClock />, borderless: true}}` or at least i think get rid of the styled component with borderless
```suggestion
        trigger={(props, isOpen) => (
          <BreadcrumbControl
            size="xs"
            icon={<IconClock />}
            showChevron={false}
            isOpen={isOpen}
            {...props}
          />
        )}
```

File: static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx (hunk 2 above)
Comment (leeandher) [reply static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx:148]: I'll account for this but I still need `allCrumbs` for the drawer content. Thanks!

File: static/app/components/events/breadcrumbs/breadcrumbsDataSection.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    expect(screen.getByText('-1min')).toBeInTheDocument();
+    expect(screen.queryByText('18:00:48')).not.toBeInTheDocument();
+  });
+
+  it.each([
+    {action: 'Search', elementRole: 'textbox'},
+    {action: 'Filter', elementRole: 'button'},
+    {action: 'Sort', elementRole: 'button'},
+  ])(
+    'opens the drawer, and focuses $action $elementRole when $action button is pressedb',
```
Comment (scttcper) [inline static/app/components/events/breadcrumbs/breadcrumbsDataSection.spec.tsx:94]:
```suggestion
    'opens the drawer, and focuses $action $elementRole when $action button is pressed',
```

**Review by scttcper — APPROVED** [review]

**Pull request merged**